packages/bsky/src/api/app/bsky/graph/getSuggestedFollowsByActor.ts

```diff
@@ -0,0 +1,152 @@
+import { Server } from '../../../../lexicon'
+import AppContext from '../../../../context'
+import { InvalidRequestError } from '@atproto/xrpc-server'
+
+const MAX_RESULTS_LENGTH = 10
+const LIKES_THRESHOLD = 50
+
+export default function (server: Server, ctx: AppContext) {
+  server.app.bsky.graph.getSuggestedFollowsByActor({
+    auth: ctx.authVerifier,
+    handler: async ({ auth, params }) => {
+      const { actor } = params
+      const viewer = auth.credentials.did
+
+      const db = ctx.db.getReplica()
+      const actorService = ctx.services.actor(db)
+      const actorDid = await actorService.getActorDid(actor)
+
+      if (!actorDid) {
+        throw new InvalidRequestError('Actor not found')
+      }
+
+      const likes = await db.db
+        .selectFrom('like')
+        .select(['subject', 'creator'])
+        .where('creator', '=', actorDid)
+        .limit(1000)
+        .execute()
+
+      let suggestions: Awaited<
+        ReturnType<typeof actorService.views.hydrateProfiles>
+      > = []
+      let allActors: any[] = []
+
+      const viewerFollows = db.db
+        .selectFrom('follow')
+        .where('creator', '=', viewer)
+        .select('subjectDid')
+
+      if (likes.length >= LIKES_THRESHOLD) {
+        // get posts to get their authors
+        const posts = await db.db
+          .selectFrom('post')
+          .where(
+            'post.uri',
+            'in',
+            likes.map((l) => l.subject),
+          )
+          .select(['creator', 'uri'])
+          .execute()
+
+        const authorDIDs = Object.values(posts).map((p) => p.creator)
+        const authorDIDsExcludingActorAndViewer = authorDIDs.filter(
+          (did) => did !== actorDid && did !== viewer,
+        )
+
+        const authorsMappedByMostCommon =
+          authorDIDsExcludingActorAndViewer.reduce((acc, did) => {
+            acc[did] = (acc[did] || 0) + 1
+            return acc
+          }, {} as Record<string, number>)
+        const authorsSortedByMostCommon = Object.entries(
+          authorsMappedByMostCommon,
+        )
+          .sort((a, b) => b[1] - a[1])
+          .slice(0, 20) // take top 20 most common
+
+        // get the profiles of the authors
+        const authors = await db.db
+          .selectFrom('actor')
+          .where(
+            'actor.did',
+            'in',
+            authorsSortedByMostCommon.map((a) => a[0]),
+          )
+          .selectAll()
+          .execute()
+        const sortedAuthors = authorsSortedByMostCommon
+          .map(([did]) => authors.find((a) => a.did === did))
+          .filter(Boolean) as typeof authors
+
+        allActors = sortedAuthors
+      } else {
+        const popularFollows = await db.db
```

Comment:
could we split each of these subqueries into it's own fn outside of the route handler (like just at the bottom of the file)?

it may make the control flow of the handler a bit easier to follow.

my initial thought when seeing this was "wow that's a lot of queries we're doing" before realizing that most of these queries aren't actually called because they're in if/elses

Reply:
Cleaned this up a bit, lmk what ya think

Reply:
ah sorry i phrased this wrong

this did clean up quite a bit & i can follow it well!

I should have said "queries" not "subqueries.

I just mean something like:
```ts
const suggestions = await getSuggestionsFromLikes(...)
if(suggestions.length < MAX) {
  const fromFollows = await getSuggestsionFromFollow(...)
  // append
} 
if(suggestions.length < MAX) {
  const fromSuggestedFollows = await getSuggestionsFromSuggestedFollows(...)
  // append
}
```

where each of those function impls are what you currently have just split out into a named function.

then the route handler itself can be like 15 lines & really easy to follow what it's doing